genie-core/src/main/java/com/netflix/genie/core/jpa/services/JpaServiceUtils.java

```diff
@@ -263,4 +277,38 @@ static JobMetadata toJobMetadataDto(final JobMetadataProjection jobMetadataProje
 
         return builder.build();
     }
+
+    private static <B extends CommonDTO.Builder, E extends BaseProjection> void setDtoMetadata(
+        final B builder,
+        final E entity
+    ) {
+        if (entity.getMetadata().isPresent()) {
+            try {
+                final String metadata = entity.getMetadata().get();
+                builder.withMetadata(metadata);
+            } catch (final GeniePreconditionException gpe) {
+                // Since the DTO can't be constructed on input with invalid JSON this should never even happen
+                log.error("Invalid JSON metadata. Should never happen.", gpe);
+                builder.withMetadata((JsonNode) null);
```

Comment:
Why catching this rather than letting it throw?

Reply:
I could throw a `GenieServerException` but that would basically fail the entire action. Since in reality it should be impossible cause the input has to be valid JSON I felt it wasn't worthwhile to have `throws` messages all over the place. If the Genie Exceptions had been moved to runtime exceptions by now I would have let it do that for sure. Otherwise I need to make it bubble up a checked exception and felt it wasn't worth it at this time given our intention to make the exceptions runtime exceptions anyway